src/utils.h

```diff
@@ -245,5 +245,23 @@ FILE * fopen_verbose(const char * const path);
  * when the environment variable doesn't exits.
  */
 void add_paths_from_env(GPtrArray *arr, char *env_name, char *subdir, char *alternative);
+
+/**
+ * Hexdumps buffer, 1 line = 8 bytes with character representation
+ *
+ * @param buffer Buffer to hexdump
+ * @param len Length of the buffer
+ */
+void hexdump(char * buffer, int len);
+
+//! Wrapper around hexdump() for string
+#define STR_HEXDUMP(STR)        hexdump(STR, strlen(STR))
+
+//! Escape LF and remove CR from string: \n -> \\n
```

Comment:
I think you shouldn't need to escape anything (it's C, not bash :p). Instead, try using NULL bytes for separation.

Reply:
The escaping is done because some fields need the escaped form (format) and some interpret it (body, summary). If I didn't  escape it, BASH would escape the unescaped half and you'd probably mess up formatting of notifications :slightly_smiling_face: